zcash_primitives/src/transaction/builder.rs

```diff
@@ -272,6 +273,30 @@ impl BuildResult {
     }
 }
 
+/// The result of [`Builder::build_for_pczt`].
+///
+/// It includes the PCZT components along with metadata describing how spends and outputs
+/// were shuffled in creating the transaction's shielded bundles.
+#[derive(Debug)]
+pub struct PcztResult<P: Parameters> {
+    pub pczt_parts: PcztParts<P>,
+    pub sapling_meta: SaplingMetadata,
+    pub orchard_meta: orchard::builder::BundleMetadata,
+}
+
+/// The components of a PCZT.
+#[derive(Debug)]
+pub struct PcztParts<P: Parameters> {
+    pub params: P,
```

Comment:
Why does the `PcztParts` need to carry the parameters along with it? This struct is created by `build_for_pczt`, and the parameters are those that were originally provided to the builder.

Given that a builder effectively a curried function, you're just giving back a value that the caller originally passed to you.

Reply:
We need to obtain `coin_type` when setting up the PCZT, and given that we already have a pattern of passing around parameters, this seemed like the most type-safe way to do so.